Bottom: 14c7bcfd24a1bb54eeb51dc3d806d2ddb4515d0d
Top:    27f0c58a00a5a686ad609bff77ae7e1ff8087a44
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 00:34:13 +0000

Refresh of work-in-progress

---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index cfc1ed4..b7f9eea 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -302,9 +302,15 @@ nogui_prompt_for (const char *msg, const char *default_string)
     printf ("%s : ", msg);
 
   s = fgets (buf, 1024, stdin);
-  if (default_string && (s == NULL || buf[0] == 0))
-    strcpy (buf, default_string);
-  return buf;
+  if (s == NULL || buf[0] == '\0' || buf[0] == '\r' || buf[0] == '\n')
+    {
+      if (default_string != NULL)
+        return strdup (default_string);
+      else
+        return strdup ("");
+    }
+  else
+    return strdup (buf);
 }
 
 /* FIXME - this could use some enhancement to actually use the other
@@ -322,10 +328,15 @@ nogui_fileselect (const char *title, const char *descr,
   else
     printf ("%s : ", title);
 
-  s = fgets (buf, 1024, stdin);
-  if (default_file && (s == NULL || buf[0] == 0))
-    strcpy (buf, default_file);
-  return buf;
+  if (s == NULL || buf[0] == '\0' || buf[0] == '\r' || buf[0] == '\n')
+    {
+      if (default_file != NULL)
+        return strdup (default_file);
+      else
+        return NULL;
+    }
+  else
+    return strdup (buf);
 }
 
 static int
